### PR TITLE
Set clang-format key to clang-tools-extra on RHEL

### PR DESCRIPTION
**Pull request by christophebedard**
Please add the following dependency to the rosdep database.

## Package name:

Set `clang-format` to `clang-tools-extra` on RHEL.

## Package Upstream Source:

N/A

## Purpose of using this:

`ros-jazzy-ament-clang-format` depends on `clang-format` on Ubuntu, but not on RHEL. The rosdep entry for `clang-format` on RHEL is `clang`, which does not provide `clang-format`. The RHEL package that provides `clang-format` is apparently `clang-tools-extra`: https://github.com/ros-tooling/setup-ros/pull/694#issuecomment-2159234767.

Since `clang-tidy` also maps to `clang-tools-extra` on RHEL and skips RHEL 7, I've done the same for `clang-format`.

## Links to Distribution Packages

- Fedora: https://packages.fedoraproject.org/
  - https://packages.fedoraproject.org/pkgs/clang/clang-tools-extra/
- rhel: https://rhel.pkgs.org/
  - https://pkgs.org/search/?q=clang-tools-extra

### CODE REVIEW

**Review by cottsay — APPROVED** [review]
Thanks, it looks like `fedora` is in the same boat. Mind fixing that as well?

https://packages.fedoraproject.org/pkgs/clang/clang-tools-extra/

**Comment by christophebedard** [comment]
> Since clang-tidy also maps to clang-tools-extra on RHEL and skips RHEL 7, I've done the same for clang-format.

Oops, I didn't commit that part. Fixed.

> Thanks, it looks like `fedora` is in the same boat. Mind fixing that as well?
>
> https://packages.fedoraproject.org/pkgs/clang/clang-tools-extra/

Done!

It looks like it might be the same for alpine too, but I can only find packages with the `clang` version in the name, e.g., `[clang18-extra-tools](https://pkgs.alpinelinux.org/package/edge/main/x86_64/clang18-extra-tools)`: https://pkgs.alpinelinux.org/packages?name=clang*tools&branch=edge&repo=&arch=x86_64&maintainer=. Unless I'm missing something, the current value for `clang-tidy` for `alpine` is `clang-extra-tools`, which doesn't exist(?).

File: rosdep/base.yaml
Comment (clalancette) [inline rosdep/base.yaml:555]: If we look at what binaries the `clang-format` tools provides on Ubuntu, it looks like it provides all of:

```
/usr/bin/clang-format
/usr/bin/clang-format-diff
/usr/bin/git-clang-format
```

On RHEL and Fedora, `clang-tools-extra` provides both `clang-format` and `clang-format-diff`, while `git-clang-format` comes in via the `git-clang-format` package.

Thus it seems to me that this should be:

```suggestion
  fedora: [clang-tools-extra, git-clang-format]
```

(the same goes for the `rhel` key below)

**Review by clalancette — CHANGES_REQUESTED** [review]

File: rosdep/base.yaml
Comment (christophebedard) [reply rosdep/base.yaml]: makes sense, done

**Review by clalancette — APPROVED** [review]
Looks good to me, but I'd like to get @cottsay 's approval again for the latest changes before merging.

**Review by cottsay — APPROVED** [review]

**Pull request merged**
